heat/core/communication.py

```diff
@@ -61,6 +61,8 @@ class MPICommunication(Communication):
         torch.int64: MPI.LONG,
         torch.float32: MPI.FLOAT,
         torch.float64: MPI.DOUBLE,
+        torch.complex64: MPI.COMPLEX,
+        torch.complex128: MPI.DOUBLE_COMPLEX,
```

Comment:
does torch have half complex?

Reply:
pytorch has complex32, but we don't support 16 bit floats

Reply:
we could add that to the comms package, its a small change. but that is not related to this PR